include/ros_alarms/listener.hpp

```diff
@@ -72,9 +81,16 @@ class AlarmListener
   // Registers a callback to be invoked when alarm is cleared
   void add_clear_cb(callable_t cb);
 
+  // Returns the last time the alarm was updated
+  ros::Time last_update_time() const { return __last_update; }
+
+  // Returns the time since last update
+  ros::Duration time_since_update() const { return ros::Time::now() - __last_update; }
+
   void clear_callbacks()      { __callbacks.clear(); }
 
 private:
+  bool __ok = true;  // listener fail-bit
```

Comment:
The user can check the ok() method which returns the value here, which would be set to false in case of a serious failure.